src/parallel/processor.py

```
@@ -406,10 +406,17 @@ def _check_pool_saturation(
         temporarily-busy pools (abandoned tasks that finish in O(timeout)) time
         to free a slot before declaring permanent saturation.
         """
-        if not pending or not all(future_started[f] is None for f in pending):
+        if not pending:
             return None
         now = time.monotonic()
-        stalled = [f for f in pending if (now - future_queued_at.get(f, now)) > timeout * 2]
+        # Only futures that have never started (future_started[f] is None) AND
+        # have been queued for longer than 2×timeout count as stalled. Futures
```

Ruff (0.15.13)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)